src/Chapter3.hs

```diff
@@ -374,6 +379,21 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Knight = Knight {
+    health :: Int,
+    attck :: Int,
+    gold :: Int
+} deriving Show
+
+data Monster = Monster {
+    health' :: Int,
+    attck' :: Int,
+    gold' :: Int
+} deriving Show
+
+-- fight :: Knight -> Monster -> Int
```

Comment:
Do you have any troubles implementing this function? I think, the easiest way to implement it is to use fields as getters and compare a few values.